applications/exposure-checker/templates/deployment.yaml

```diff
@@ -0,0 +1,104 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Chart.Name }}
+  labels:
+    {{- include "exposure-checker.labels" . | nindent 4 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- include "exposure-checker.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      labels:
+        {{- include "exposure-checker.selectorLabels" . | nindent 8 }}
+    spec:
+      volumes:
+        - name: "butler-secrets-raw"
+          secret:
+            secretName: {{ include "exposure-checker.fullname" . }}
+        - name: "butler-secrets"
+          emptyDir:
+            sizeLimit: 50Mi
+        {{- with .Values.config.volumes }}
+        {{- . | toYaml | nindent 8 }}
+        {{- end }}
+      containers:
+      - name: {{ .Chart.Name }}
+        image: {{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion}}
+        imagePullPolicy: Always
+        env:
+{{- range $key, $value := .Values.environment }}
+          - name: {{ $key | quote }}
+            value: {{ $value | quote }}
+{{- end }}
+          - name: S3_KEY
+            valueFrom:
+              secretKeyRef:
+                name: {{ include "exposure-checker.fullname" . }}
+                key: S3_KEY
+          - name: S3_SECRET
+            valueFrom:
+              secretKeyRef:
+                name: {{ include "exposure-checker.fullname" . }}
+                key: S3_SECRET
+          - name: BUTLER_REPO
+            value: {{ .Values.config.butler_repo }}
+          - name: BUTLER_COLLECTION
+            value: {{ .Values.config.butler_collection }}
+          - name: S3_PROFILE_NAME
+            value: {{ .Values.config.s3_profile_name }}
+          - name: S3_ENDPOINT_URL
+            value: {{ .Values.config.s3_endpoint_url }}
+          - name: DB_HOST
+            value: {{ .Values.config.db.hostname }}
+          - name: DB_USERNAME
+            value: {{ .Values.config.db.username }}
+          - name: DB_PASSWORD
+            valueFrom:
+              secretKeyRef:
+                name: {{ include "exposure-checker.fullname" . }}
+                key: databasePassword
+          - name: DB_DBNAME
+            value: {{ .Values.config.db.db_name }}
+        resources:
+          {{- toYaml .Values.resources | nindent 10 }}
+        {{- if .Values.config.envFromSecretPath }}
+        envFrom:
+          - secretRef: {{ include "exposure-checker.fullname" . }}
+        {{- end }}
+        {{- if .Values.config.volume_mounts }}
+        volumeMounts:
+          {{- with .Values.config.volume_mounts }}
+          {{- . | toYaml | nindent 10 }}
+          {{- end }}
+        {{- end }}
+      initContainers:
+        # To deal with the Postgres file permission issues,
+        # copy the secrets from butler-secrets-raw to butler-secrets.
+        - name: fix-secret-permissions
```

Comment:
In the app, use `lsst-utils>=29.2025.1500` which supports reading butler auth secrets from the environment and you don't have to do this initContainer dance.

Reply:
I am excited to try this out.